store/mongo/migration_2_0_0.go

```diff
@@ -0,0 +1,152 @@
+	mstorev1 "github.com/mendersoftware/go-lib-micro/store"
+	mstore "github.com/mendersoftware/go-lib-micro/store/v2"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	mopts "go.mongodb.org/mongo-driver/mongo/options"
+)
+
+const findBatchSize = 100
+
+type migration_2_0_0 struct {
+	ms  *DataStoreMongo
+	ctx context.Context
+}
+
+func (m *migration_2_0_0) Up(from migrate.Version) error {
+	id := identity.FromContext(m.ctx)
+	if id != nil && id.Tenant != "" {
+		return nil
+	}
+
+	ctx := context.Background()
+	client := m.ms.client
+	dbs, err := migrate.GetTenantDbs(ctx, client, mstorev1.IsTenantDb(DbName))
+	if err != nil {
+		return err
+	}
+	dbs = append([]string{DbName}, dbs...)
+
+	collections := map[string]struct {
+		Indexes []mongo.IndexModel
+	}{
+		DbDevicesColl: {
+			Indexes: []mongo.IndexModel{
+				{
+					Keys: bson.D{
+						{Key: mstore.FieldTenantID, Value: 1},
+						{Key: DbDevId, Value: 1},
+					},
+					Options: mopts.Index().
+						SetName(mstore.FieldTenantID + "_" + DbDevId),
+				},
+				{
+					Keys: bson.D{
+						{Key: mstore.FieldTenantID, Value: 1},
+						{Key: DbDevUpdatedTs, Value: 1},
+					},
+					Options: mopts.Index().
+						SetName(mstore.FieldTenantID + "_" + DbDevUpdatedTs),
+				},
+			},
+		},
+	}
+
+	// for each collection...
+	for collection, config := range collections {
+		coll := client.Database(DbName).Collection(collection)
+		// drop all the existing indexes, ignoring the errors
+		_, _ = coll.Indexes().DropAll(ctx)
+
+		// create the new indexes
+		_, err := coll.Indexes().CreateMany(ctx, config.Indexes)
+		if err != nil {
+			return err
+		}
+	}
+
+	// for each database...
+	for _, db := range dbs {
+		tenantID := mstorev1.TenantFromDbName(db, DbName)
+		ctx := identity.WithContext(ctx, &identity.Identity{
+			Tenant: tenantID,
+		})
+		// for each collection...
+		for collection, _ := range collections {
+			// get all the documents in the collection
+			findOptions := mopts.Find().
+				SetBatchSize(findBatchSize).
+				SetSort(bson.D{{Key: DbDevId, Value: 1}})
+			coll := client.Database(db).Collection(collection)
+			cur, err := coll.Find(ctx, bson.D{}, findOptions)
+			if err != nil {
+				return err
+			}
+			defer cur.Close(ctx)
+
+			writes := make([]mongo.WriteModel, 0, findBatchSize)
+
+			// migrate the documents
+			for cur.Next(ctx) {
+				item := bson.D{}
+				err := cur.Decode(&item)
+				if err != nil {
+					return err
+				}
+
+				item = mstore.WithTenantID(ctx, item)
+				if db == DbName {
+					filter := bson.D{}
+					for _, i := range item {
+						if i.Key == DbDevId {
+							filter = append(filter, i)
+						}
+					}
+					writes = append(writes, mongo.NewReplaceOneModel().SetFilter(filter).SetReplacement(item))
+				} else {
+					writes = append(writes, mongo.NewInsertOneModel().SetDocument(item))
+				}
+				if len(writes) == findBatchSize {
+					_, err := client.Database(DbName).Collection(collection).BulkWrite(ctx, writes)
```

Comment:
```suggestion
					_, err := coll.BulkWrite(ctx, writes)
```

Reply:
In my [first comment](https://github.com/mendersoftware/inventory/pull/261#discussion_r639614075) I intended to replace the `InsertOne/ReplaceOne` operations with `InsertMany/UpdateMany`, but I'm not sure if there's any difference in terms of performance using `BulkWrite`.

Reply:
This suggestion is wrong, because we don't write to `coll` (the input collection), but to the collection in the global database. Nevertheless, I created a variable and reused it.

Reply:
Regarding performances, InsertMany/UpdateMany use BulkWrite behind the scenes.